cs/__init__.py

```diff
@@ -0,0 +1,143 @@
+def cs_types_encode(key):
+    try:
+        cs_type = _cs_types_encode[key]
+    except KeyError:
+        cs_type = _cs_types_common[key]
+    if cs_type == "NA":
+        raise NotImplementedError("Missing type Mapping")
+    return cs_type
+
+
+def cs_types_decode(key):
+    try:
+        cs_type = _cs_types_decode[key]
+    except KeyError:
+        cs_type = _cs_types_common[key]
+    if cs_type == "NA":
+        raise NotImplementedError("Missing type Mapping")
+    return cs_type
+
+
+def cs_escape_keyword(value):
+    keywords = ["base"]
+    if value not in keywords:
+        return value
+    return "@" + value
+
+
+_cs_types_common = {
+    "boolean": "bool",
+    "int": "int",
+    "long": "long",
+    "byte": "byte",
+    "Integer": "int",
+    "Long": "long",
+    "UUID": "Guid",
+    "enum": "int",
+
+    "longArray": "long[]",
+    "byteArray": "byte[]",
+    "String": "string",
+    "Data": "IData",
+
+    "Address": "Hazelcast.IO.Address",
+    "ErrorHolder": "Hazelcast.Client.Protocol.ErrorHolder",
+    "StackTraceElement": "Hazelcast.Util.StackTraceElement",
+    "SimpleEntryView": "Hazelcast.Map.SimpleEntryView<IData, IData>",
+    "RaftGroupId": "NA",
+    "WanReplicationRef": "NA",
+    "HotRestartConfig": "NA",
+    "EventJournalConfig": "NA",
+    "MerkleTreeConfig": "NA",
+    "TimedExpiryPolicyFactoryConfig": "NA",
+    "MapStoreConfigHolder": "NA",
+    "QueueStoreConfigHolder": "NA",
+    "RingbufferStoreConfigHolder": "NA",
+    "NearCacheConfigHolder": "NA",
+    "EvictionConfigHolder": "NA",
+    "NearCachePreloaderConfig": "NA",
+    "PredicateConfigHolder": "NA",
+    "DurationConfig": "NA",
+
+    "MergePolicyConfig": "NA",
+    "CacheConfigHolder": "NA",
+    "CacheEventData": "NA",
+    "QueryCacheConfigHolder": "NA",
+    "DistributedObjectInfo": "Hazelcast.Client.DistributedObjectInfo",
+    "IndexConfig": "Hazelcast.Config.IndexConfig",
+    "AttributeConfig": "NA",
+    "ListenerConfigHolder": "NA",
+    "CacheSimpleEntryListenerConfig": "NA",
+
+    "Map_String_String": "IDictionary<string, string>"
+}
+
+_cs_types_encode = {
+    "CacheEventData": "NA",
```

Comment:
Some types like `CacheEventData`, `List_Xid` ... marked as `NA` in both `_cs_types_encode ` and `_cs_types_decode`. Maybe we can move them to `_cs_types_common` so the same things are not written twice.

Reply:
The reason is still there for not being in common, they'll need an encode and decode mappings in future but they are just not implemented. So let's keep it as is.